src/controllers/user.js

```diff
@@ -156,6 +161,12 @@ export const updateUserById = async (req, res) => {
     return sendDataResponse(res, 404, { id: 'User not found' })
   }
 
+  if (req.user.role !== 'ADMIN') {
```

Comment:
Does this prevent non-admin users from updating their account details?

Reply:
Yes, this prevents non-admin users from updating their account role.

Reply:
Where does it check if they are trying to update their account role? From what I can see this will stop regular users from updating any field

Reply:
I created an admin.js file in controllers folder and I moved all my logic in there so as not to interfere with the updateUserById function in user.js